<launch>
  <arg name="is_low_level" default="0"/>
  <arg name="torque_safe_limit" default="0"/>
  <arg name="udp_connection" default="0"/>

  <include file="$(find be2r_cmpc_unitree)/launch/set_dyn_params.launch" />

  <param name = "robot_description" command = "$(find xacro)/xacro --inorder '$(find raisim_unitree_ros_driver)/description/urdf/a1_edited.urdf' " />

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

  <node pkg="be2r_cmpc_unitree" type="unitree_ctrl" name="unitree_ctrl" output="screen" launch-prefix="bash -c 'sleep 1.0; $0 $@' ">
    <param name="map_topic" value="/elevation_mapping/elevation_map"/>
    <param name="is_low_level" type="bool" value="$(arg is_low_level)" />
    <param name="torque_safe_limit" type="bool" value="$(arg torque_safe_limit)" />
    <param name="udp_connection" type="bool" value="$(arg udp_connection)" />
  </node>

  <node pkg="tf" type="static_transform_publisher" name="world_to_odom" args="0 0 0 0 0 0 world odom 100" />

</launch> 
